Automattic/jetpack
# Update/my jetpack update interstitial pages

This PR mainly updates the Interstitial pages using the `<ConnectedProductOffer />` component, recently added. Also, updates the interstitial making the component more robust and simpler:
- removes the `bundle` property. It's easily pickable from the `isUpgradableByBundle`.
- Define the secondary section automatically for not upgradable products.

⚠️ **IMPORTANT** ⚠️
This changes are touching the Interstitial pages which are currently used in My Jetpack dashboard, meaning it can break the app 🤞 

Fixes #

#### Changes proposed in this Pull Request:

* Replace Interstitial pages using the ConnectedProductOffer component instead of the ProductDetailCard
* Delete ProductDetailCard component
* Adjust Interstitial component improving the implementation

#### Jetpack product discussion


p1649860877787919-slack-C02TQF5VAJD

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






**Testing with Storybook**
* Go to My Jetpack / Interstitial page
* Use only the default story
* Check the pages look rightly.

https://user-images.githubusercontent.com/77539/163436982-11b1473a-0efd-4c83-8f9f-3c07fd93d3a0.mov


**Testing with My Jetpack overview**
* Check the interstitial pages look properly after these changes
  * `http://<your-testing-site>/wp-admin/admin.php?page=my-jetpack#/add-anti-spam`
  * `http://<your-testing-site>/wp-admin/admin.php?page=my-jetpack#/add-backup`
  * `http://<your-testing-site>/wp-admin/admin.php?page=my-jetpack#/add-boost`
  * so on...
* Check the whole process of adding the product works as expected

  * `http://<your-testing-site>/wp-admin/admin.php?page=my-jetpack#/add-anti-spam`
